### DISCUSSION TITLE
Upgrading b2evolution 7.2.5 for PHP 7.x to PHP 8.0 with Rector

### DISCUSSION THREAD

**Discussion by Metztli** [rectorphp/rector#7642] [General]
# Bug Report

| Subject        | Details                                                         |
| :b2evolution 7.2.5 partial upgrade from 7.4 to 8.0 | :Omits some variables in original PHP 7.x code install file(s) and ... |
| Rector version | Rector 0.14.8      |

Installed rector with Composer, created rector.php with content:

```
<?php

declare(strict_types=1);

use Rector\CodeQuality\Rector\Class_\InlineConstructorDefaultToPropertyRector;
use Rector\Config\RectorConfig;
use Rector\Set\ValueObject\LevelSetList;

return static function (RectorConfig $rectorConfig): void {
        $rectorConfig->import(LevelSetList::UP_TO_PHP_80);

};

```

## Minimal PHP Code Causing Issue
PHP 7.4.33 (Debian 11 Bullseye) powering Rector

## Expected Behaviour

I read wonders, literally, about how rector smoothly upgrades PHP 7.x code to 8.0, i.e., [Automatically upgrading your code to PHP 8 using Rector](https://spatie.be/courses/front-line-php/automatically-upgrading-your-code-to-php-8-using-rector) , as I was looking for a way to upgrade [b2evolution](https://github.com/b2evolution/b2evolution) code for PHP 7.x into PHP 8.1 . Since Rector spectacularly failed in the upgrade, I downgraded the upgrade requirement to PHP 8.0. I am not a PHP developer per se but more of a hacker who fixes others issues by inductive reasoning. 

After several Rector iterations in an eight(8) CPU sixteen(16 threads) AMD server I got Rector to complete the upgrade.
Notwithstanding, upon testing the b2evolution installation I came about several missing variables in the Rector code. Additionally, a whole set of statements was missing from a function in the installer files. I painstakingly compared some of the old and upgraded install files and manually added the omissions.

Subsequently tested the b2evolution 7.2.5 installer, the user provided database values, i.e., username, database name, password, are not upgraded by the resulting PHP 8.0 code. If, however, I manually modify those values in the relevant configuration file the b2evolution installation (without demo content) proceeds and completes, apparently successfully; until the time when I attempt to login with the one-time provided password fails during multiple login attempts. It seems the Rector -generated PHP 8.0 code fails to access properly the database.

**Comment by eglacias** [comment]
This person did a lot of work, it sure would be useful to be able to upgrade b2evolution properly.

Seems like this would be a good test for making Rector an even better piece of software?
